Migrate modules from pint-server code

smarlowucf opened the pull request:
- Migrate modules from pint server code
- Fix license

Note: pint_models/database.py and pint_models/models.py are pulled directly from pint server code https://github.com/SUSE-Enceladus/public-cloud-info-service/tree/master/pint_server.

smarlowucf committed:
Migrate modules from pint-server code

rjschwei commented on package/python-pint-models.spec:
```diff
@@ -0,0 +1,57 @@
+#
+
+%define upstream_name pint-models
+%define python python
```
This define is not needed, we are not going to build this for older versions of SLE.

rjschwei commented on pint_models/database.py:
```diff
@@ -0,0 +1,210 @@
+from pint_models.models import Base
+
+
+def get_environ_or_bust(key_name):
```
No cute names please, `get_environ_entry` will do just fine.

rjschwei commented on pint_models/database.py:
```diff
@@ -0,0 +1,210 @@
+
+
+def get_environ_or_bust(key_name):
+    assert key_name in os.environ, 'Environment variable %s is required.' % (
```
Should probably throw and exception so it can be handled gracefully.

rjschwei commented on pint_models/database.py:
```diff
@@ -0,0 +1,210 @@
+    db_logger.setLevel(db_logger_log_level)
+
+
+def _create_postgres_url(db_user, db_password, db_name, db_host,
```
It be nice to move the internal functions to the end rather than having them intermixed with the "public" functions.

rjschwei commented on pint_models/database.py:
```diff
@@ -0,0 +1,210 @@
+        engine_url = create_postgres_url_from_env()
+
+    # TODO(rtamalin): Remove this try/except hackery once we move forward
+    # to being based on SLE 15 SP3 or later.
```
Per comment this can be cleaned up, we are not going to build/use this on anything older the 15 SP6

rjschwei commented on setup.py:
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
```
Would it make sense to not create a setup.py file and create a toml file for this form the get go?

smarlowucf replied:
Yeah, can migrate it to toml. Although, not sure if we can entirely escape setup.py yet...

rjschwei reviewed (changes requested):
We will need to have a license discussion.

smarlowucf committed:
Revert license to Apache 2.0

smarlowucf committed:
Switch to pyproject.toml

- Rename get_environ_or_bust to get_environ_entry
- Cleanup spec file
- Drop exception handling for old versions of SLES

smarlowucf commented:
Made updates per comments

smarlowucf committed:
Set limit on SQLAlchemy version

Version 2.0.0 drops convert_unicode option

apozsuse reviewed (approved):
LGTM :+1:

smarlowucf committed:
Add requirement on pyscopg plugin for postgres

smarlowucf committed:
Remove unused requirements files

smarlowucf commented:
Going ahead with merge. Any other comments can be handled in a new PR
